statsmodels/tools/tools.py

```diff
@@ -506,7 +506,7 @@ def fullrank(x, r=None):
     if r is None:
         r = np.linalg.matrix_rank(x)
```

Comment:
This call is wasteful since it is performing a second svd.

Reply:
this function is not used, old code

We keep some of the old linalg functions as recipes, until we have our own toolset for linalg